### PR TITLE
SW-4514 Support non-localizable enums

### PR DESCRIPTION
**Pull request by sgrimm**
Some of our database-backed enums are only used internally, never shown directly
to users. Previously, we still had to add their display names to our strings table
(and those names would still be submitted for translation) because the generated
code would try to look them up.

Pull the localization functionality out of `EnumFromReferenceTable` into a new
interface `LocalizableEnum`. Enums that are exposed in the search API are required
to be localizable since they can potentially be exported in CSV files.

Whether or not an enum is localizable is controlled by an optional parameter to
`EnumTable` in the code generator config.

### CODE REVIEW

**Comment by sgrimm** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1529** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1529?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #1530** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1530?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #1531** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1531?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/1531?utm_source=stack-comment).

**Comment by sgrimm** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1529** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1529?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #1530** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1530?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #1531** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1531?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/1531?utm_source=stack-comment).

**Comment by sgrimm** [comment]
Once this is merged, I'll remove the strings in question from Phrase so it doesn't try to add them back to the enum properties files.

File: src/main/kotlin/com/terraformation/backend/db/LocalizableEnum.kt [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+          setDefaultEncoding("UTF-8")
+        }
+
+    fun <T : LocalizableEnum<T>> loadLocalizedDisplayNames(
+        locale: Locale,
+        values: List<T>
+    ): Map<T, String> {
+      val enumClass = values.first().javaClass
+      val enumName = enumClass.simpleName
+      val packageName = enumClass.packageName.substringAfterLast('.')
```
Comment (nickgraz) [inline src/main/kotlin/com/terraformation/backend/db/LocalizableEnum.kt:28]: Kotlin has everything you need!

**Review by karthikbtf — APPROVED** [review]
Looks good!

**Review by nickgraz — APPROVED** [review]

**Comment by sgrimm** [comment]
### Merge activity

* **Nov 29, 6:58 PM**: @sgrimm started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/1531).
* **Nov 29, 6:59 PM**: [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/1531) rebased this pull request as part of a merge.
* **Nov 29, 7:00 PM**: @sgrimm merged this pull request with [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/1531).

**Pull request merged**